### PR TITLE
Polish date-picker component

### PR DESCRIPTION
**Pull request by jasmussen**
This PR does a few things:

1. It refines the buttongroup component slightly. There was a small issue with the button group used for gradient type picking, that this addresses.
2. It refactors the am/pm selector in the date component to use a ButtonGroup.
3. It polishes the padding of the datetime picker, closing #20774.

Screens:

Screenshot 2020-03-12 at 11 27 54

Screenshot 2020-03-12 at 11 27 59

Screenshot 2020-03-12 at 11 28 03

Screenshot 2020-03-12 at 11 28 13

### CODE REVIEW

File: packages/components/src/date-time/style.scss [hunk 1]
```diff
@@ -7,6 +7,10 @@
 .components-datetime {
 	padding: $grid-unit-20;
 
+	.components-popover__content & {
```
Comment (jasmussen) [inline packages/components/src/date-time/style.scss]: I recall having a conversation about keeping the padding behavior isolated to the context. But this feels a decent way to tune the behavior. Correct me if I'm wrong, Riad.

**Comment by jasmussen** [comment]
This is the "before":

Screenshot 2020-03-12 at 11 39 04

File: packages/components/src/date-time/style.scss (hunk 1 above)
Comment (youknowriad) [reply packages/components/src/date-time/style.scss]: Well ideally the component works the same in a popover or outside a popover. This is to say it shouldn't have padding IMO but I do remember the same discussion previously and you said that the calendar styles (third-party) kind of forces us to do have that padding somehow?

Comment (jasmussen) [reply packages/components/src/date-time/style.scss]: Yes that was the case due to how the month pagination works. However right at this moment I can't reproduce. Let me try a thing.

Comment (jasmussen) [reply packages/components/src/date-time/style.scss]: The issue we had back then does not appear to be an issue anymore, so I've pushed a different fix in https://github.com/WordPress/gutenberg/pull/20824/commits/c7302b7971f2dee277df4ec2732606e6deb5189c

And things look right:

Screenshot 2020-03-12 at 12 02 13

Screenshot 2020-03-12 at 12 02 04

File: packages/components/src/date-time/style.scss [hunk 2]
```diff
@@ -118,51 +118,15 @@
...
-			border-color: $dark-gray-100;
-			box-shadow: inset 0 2px 5px -3px $dark-gray-500;
-			&:focus {
-				box-shadow:
-					inset 0 2px 5px -3px $dark-gray-500,
-					0 0 0 1px $white,
-					0 0 0 3px $blue-medium-focus;
-			}
-		}
-
```
Comment (youknowriad) [inline packages/components/src/date-time/style.scss]: What's all this red about :)

File: packages/components/src/date-time/time.js [hunk 3]
```diff
@@ -318,24 +319,22 @@ class TimePicker extends Component {
 							/>
 						</div>
 						{ is12Hour && (
-							<div className="components-datetime__time-field components-datetime__time-field-am-pm">
+							<ButtonGroup className="components-datetime__time-field components-datetime__time-field-am-pm">
```
Comment (youknowriad) [inline packages/components/src/date-time/time.js]: Oh that's the red thing. Cool to reuse components. 👍

**Review by youknowriad — APPROVED** [review]
Nice PR

File: packages/components/src/date-time/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/components/src/date-time/style.scss]: All that red duplicated the rules from the ButtonGroup component. In all my testing they are unnecessary with the refactor to use the actual ButtonGroup (and to toggle the selected button to primary in the ternery). You can recognize the margin-left: -1px from the button group, where it's used to ensure their borders stack. The z-index to ensure focus is uncropped is also there.

Comment (jasmussen) [reply packages/components/src/date-time/style.scss]: The border radii for the am/pm buttons are even replaced by first-child/last-child rules in the buttongroup.

**Comment by jasmussen** [comment]
Thanks for the review. I see some test failures I have to update.

**Comment by paaljoachim** [comment]
I just tested...
Clicking the Calendar help link to view the information then clicking outside the box (into empty space) it does not close.

Clicking the close text brings one back to the calendar. 
Clicking outside (into empty space) does not close the calendar box.

Screen Shot 2020-03-24 at 09 43 34
 
Date-Picker-screen

@jasmussen

**Comment by jasmussen** [comment]
Indeed, excellent catch. Mind opening a new ticket for that? I don't _think_ it's a result of this PR specifically, but feel free to point to it.

**Pull request merged**